core/commands/get.go

```diff
@@ -64,14 +60,34 @@ may also specify the level of compression by specifying '-l=<1-9>'.
 			res.SetError(err, cmds.ErrNormal)
 			return
 		}
-
 		p := path.Path(req.Arguments()[0])
+		ctx := req.Context()
+		dn, err := core.Resolve(ctx, node, p)
+		if err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+
 		var reader io.Reader
-		if archive, _, _ := req.Option("archive").Bool(); !archive && cmplvl != gzip.NoCompression {
-			// only use this when the flag is '-C' without '-a'
-			reader, err = getZip(req.Context(), node, p, cmplvl)
+		if archive, _, _ := req.Option("archive").Bool(); !archive {
+			if cmplvl != gzip.NoCompression {
+				// only use this when the flag is '-C' without '-a'
+				reader, err = utar.DagArchive(ctx, dn, p.String(), node.DAG, false, cmplvl)
+			} else {
+				outPath, _, _ := req.Option("output").String()
+				if len(outPath) == 0 {
+					_, outPath = gopath.Split(p.String())
+					outPath = gopath.Clean(outPath)
+				}
+
+				if err := utar.DagFS(ctx, dn, node.DAG, outPath); err != nil {
+					res.SetError(err, cmds.ErrNormal)
+					return
+				}
+				return
+			}
```

Comment:
:( i moved all this stuff out of this function and into its own because it was very confusing here.